arwen/elrondapi/elrondei.go

```diff
@@ -914,6 +920,28 @@ func v1_4_getESDTTokenData(
 	return int32(len(esdtData.Value.Bytes()))
 }
 
+//export v1_4_getESDTLocalRoles
+func v1_4_getESDTLocalRoles(context unsafe.Pointer, tokenIdHandle int32) int64 {
+	managedType := arwen.GetManagedTypesContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+	storage := arwen.GetStorageContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	gasToUse := metering.GasSchedule().ElrondAPICost.StorageLoad
+	metering.UseGasAndAddTracedGas(storageLoadLengthName, gasToUse)
+
+	tokenID, err := managedType.GetBytes(tokenIdHandle)
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return 0
```

Comment:
can the final `return int64(len(data))` be 0? If so, then change the return value on this `if arwen.WithFault()`

Reply:
0 is considered NONE, and the reasoning is that if it is not valid it doesn't have roles

Reply:
better return -1 on `if arwen.WithFault`, not 0